shtc1/shtc1.c

```diff
@@ -64,85 +63,30 @@ static const uint8_t SHTC1_ADDRESS = SHT_ADDRESS;
 static const uint8_t SHTC1_ADDRESS = 0x70;
 #endif
 
-static const uint16_t SHTC1_PRODUCT_CODE_MASK = 0x001F;
-static const uint16_t SHTC1_PRODUCT_CODE = 0x0007;
-static const uint16_t SHTC3_PRODUCT_CODE_MASK = 0x083F;
-static const uint16_t SHTC3_PRODUCT_CODE = 0x0807;
-
 static uint16_t shtc1_cmd_measure = SHTC1_CMD_MEASURE_HPM;
 
-/**
- * PM_SLEEP is equivalent to
- * if (ret) {
- *     (void)shtc1_sleep(); // attempting to sleep but ignore return value
- * } else {
- *     (ret) = shtc1_sleep();
- * }, (ret)
- */
-#define PM_SLEEP(ret) ((ret) ? (shtc1_sleep(), (ret)) : ((ret) = shtc1_sleep()))
-
-/**
- * PM_WAKE wakes the sensor if the command is successful (ret == 0).
- * The macro code is equivalent to:
- * (ret) = shtc1_wakeup();
- * if ((ret) == 0) {
- *     (ret) = (cmd);
- *     if ((ret) != 0)
- *          shtc1_sleep(); // cmd failed, go back to sleep
- * }, (ret)
- */
-#define PM_WAKE(ret, cmd)                                                      \
-    (((ret) = shtc1_wakeup())                                                  \
-         ? (ret) /* ret = STATUS_WAKEUP_FAILED */                              \
-         : (((ret) = (cmd))                                                    \
-            ? shtc1_sleep(),                                                   \
-            (ret) /* ret = cmd failed (ret != 0),                              \
-                     sensor potentially asleep */                              \
-            : (ret)) /* ret = STATUS_OK and sensor is awake */)
-
-static uint8_t supports_sleep = 1;
-static uint8_t sleep_enabled = 1;
-
-static int16_t shtc1_sleep() {
-    int16_t ret;
-
-    if (!supports_sleep || !sleep_enabled)
-        return STATUS_OK;
-
-    ret = sensirion_i2c_write_cmd(SHTC1_ADDRESS, SHTC3_CMD_SLEEP);
-    if (ret != STATUS_OK)
-        return STATUS_SLEEP_FAILED;
-    return STATUS_OK;
+int16_t shtc1_sleep() {
```

Comment:
I find the naming a bit confusing, since the function won't work with an `shtc1`. Should we rename the common functions to `shtcx_` and the `shtc3` only ones with `shtc3_`?

Reply:
Fair point. I actually considered that, but since - arguably - the function actually works (even though it will fail), I think in the end there's less potential for confusion this way. I'd reason that people who own an SHTC3 would be more confused by the fact that they have to call `shtc1_` functions in some cases and `shtc3_` functions in other cases.

Reply:
> 'd reason that people who own an SHTC3 would be more confused by the fact that they have to call shtc1_ functions in some cases and shtc3_ functions in other cases.

That's why I'd rename the `shtc1_` ones  to `shtcx_`.